src/readblock.c

```diff
@@ -0,0 +1,299 @@
+
+    F->tags = tags;
+    F->n_tags = n_tags;
+    return F;
+}
+
+
+/***************************************************************************************/
+static ffio *ffio_destroy (ffio *F) {
+/****************************************************************************************
+
+****************************************************************************************/
+    free (F->args);
+    free (F->next_args);
+    free (F);
+    return 0;
+}
+
+
+/***************************************************************************************/
+static int at_decorative_element (ffio *F) {
+/****************************************************************************************
+    A decorative element consists of a line of at least 5 consecutive identical chars,
+    starting at buffer position 0:
+    "-----", "=====", "*****", etc.
+
+    A decorative element serves as a end delimiter for the current element, and
+    continues until a gie command verb is found at the start of a line
+****************************************************************************************/
+    int i;
+    char *c;
+    if (0==F)
+        return 0;
+    c = F->next_args;
+    if (0==c)
+        return 0;
+    if (0==c[0])
+        return 0;
+    for (i = 1; i < 5; i++)
+        if (c[i]!=c[0])
+            return 0;
+    return 1;
+}
+
+
+/***************************************************************************************/
+static const char *at_tag (ffio *F) {
+/****************************************************************************************
+    A start of a new command serves as an end delimiter for the current command
+****************************************************************************************/
+    size_t j;
+    for (j = 0;  j < F->n_tags;  j++)
+        if (strncmp (F->next_args, F->tags[j], strlen(F->tags[j]))==0)
+            return F->tags[j];
+    return 0;
+}
+
+
+/***************************************************************************************/
+static int at_end_delimiter (ffio *F) {
+/****************************************************************************************
+    An instruction consists of everything from its introductory tag to its end
+    delimiter.  An end delimiter can be either the introductory tag of the next
+    instruction, or a "decorative element", i.e. one of the "ascii art" style
+    block delimiters typically used to mark up block comments in a free format
+    file.
+****************************************************************************************/
+    if (F==0)
+        return 0;
+    if (at_decorative_element (F))
+        return 1;
+    if (at_tag (F))
+        return 1;
+    return 0;
+}
+
+
+/***************************************************************************************/
+static int nextline (ffio *F) {
+/****************************************************************************************
+    Read next line of input file
```

Comment:
Describe that the return is a boolean and what true and false represent

Reply:
Done